plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -872,15 +1079,18 @@ def get_deploy_transaction(  # pylint: disable=arguments-differ
                     "maxPriorityFeePerGas": max_priority_fee_per_gas,
                 }
             )
-        else:
-            gas_price = (
-                self._try_get_gas_price(gas_price_strategy)
-                if gas_price is None
-                else gas_price
-            )
+
+        if gas_price is not None:
+            transaction.update({"gasPrice": gas_price})
+
+        if gas_price is None and max_fee_per_gas is None:
+            gas_price = self._try_get_gas_price(
```

Comment:
```suggestion
            gas_pricing = self._try_get_gas_pricing(
```
otherwise it's confusing as `gas_price` is a parameter

Reply:
The same logic exists in two places, please find and fix